### ISSUE TITLE
Babel does not compile sources after upgrade of babel-preset-react-app

### ISSUE THREAD

**Issue by kirill-konshin** [kirill-konshin/react-router-redux-middleware#3]
As stated in https://github.com/kirill-konshin/create-react-server/pull/2#issuecomment-310400031 Babel stopped compiling after upgrade of babel-preset-react-app.

Potentially related to https://github.com/facebookincubator/create-react-app/issues/2524.

**Comment by kirill-konshin** [comment]
@diligiant please try the latest master.

**Comment by diligiant** [comment]
@kirill-konshin I tested both 0.2.0 and your patch retrofitted to 0.3.0 (I had to tweak things a little bit as `ModuleScopePlugin` prevents from referencing `../../../wrapper`) and it works fine. Thank you.

I don't know yet if the "es2015" side-effect on the react-project itself is harmless. I'll watch out.

**Comment by diligiant** [comment]
As we're using css-modules, we will probably need to inject options into babel-register (I'll submit a pull request). Giving it a try,
```
require('babel-register')({
        presets: [ 
            "es2015",
            "react-app"
        ]
    });
```
removes the need to change the cra project presets.

**Comment by kirill-konshin** [comment]
Babel register takes that from `.babelrc` or `babel` section of `package.json`, there's no need to call it explicitly. I will update the `0.3.0` branch today.

**Comment by diligiant** [comment]
Going further, babel requirements for cra and babel for server-cra are « opposite » so it's gonna take a lot more than just using your script (which works great btw). I see why cra team doesn't want to be involved in this right now ;)

**Comment by kirill-konshin** [comment]
I hope at some point they will at least merge my PR about the docs :)

I can't talk on their behalf but it seems they stay away from everything non-standard: CSS modules, preprocessors, annotations, so on and so on. They want to keep CRA as small as possible and deliver as much as they can at the same time.

**Comment by diligiant** [comment]
And that's good if you ask me. Having ejected, I maintain vanilla ejected cra so I can diff and apply to our stuff.

**Comment by kirill-konshin** [comment]
Released 0.2.1.

0.3.0 branch will be updated today, and probably will be released next week (w/o async route support though, need to investigate it more).

**Comment by kirill-konshin** [comment]
0.3.0 was rebased.
